csharp/src/iceboxnet/ServiceManager.cs

```diff
@@ -305,18 +238,17 @@ internal async Task<int> RunAsync()
             }
             finally
             {
-                StopAll();
+                await StopAllAsync();
```

Comment:
I think I would get rid of StopAllAsync and just include the code here given that we don't have StartAllAsync.

Reply:
Should we catch OperationCancelledException to avoid printing out an error in the log when the activation is cancelled when Ctrl-C is pressed?

Reply:
Fixed as suggested. Also added:
```
await _communicator.ShutdownAsync(); // often no-op since the communicator is already shut down.
```

at the start of the finally block to ensure the communicator is indeed shut down, even when an exception occurs.